driver/CMakeLists.txt

```diff
@@ -30,6 +30,7 @@ find_package(Threads REQUIRED)
 add_executable(MIOpenDriver main.cpp InputFlags.cpp)
 target_link_libraries(MIOpenDriver MIOpen)
 target_link_libraries(MIOpenDriver ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(MIOpenDriver ${CMAKE_DL_LIBS})
```

Comment:
@JehandadKhan  it is happening when linking with all executables, not just MIOpenDriver.
ref: http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/jd%2Flink_dl/1/pipeline

Reply:
and ${CMAKE_THREAD_LIBS_INIT} might also be needed for `-lpthread`